integration/filter_test.go

```diff
@@ -103,4 +105,53 @@ var _ = Describe("Filter", func() {
 			Ω(session).Should(gbytes.Say(`onepkg: \["beluga", "bird", "cat", "chicken", "cow", "dog", "giraffe", "koala", "monkey", "otter", "owl", "panda"\]`))
 		})
 	})
+
+	Describe("Semantic Version Filtering", func() {
+		BeforeEach(func() {
+			fm.MountFixture("semver")
+		})
+
+		It("filters specs based on semantic version constraints", func() {
+			session := startGinkgo(filepath.Join(fm.TmpDir, "semver"),
+				"--sem-ver-filter=2.2.0",
+				"--json-report=report.json",
+			)
+			Eventually(session).Should(gexec.Exit(0))
+			specs := Reports(fm.LoadJSONReports("semver", "report.json")[0].SpecReports)
+			passedSpecs := []string{
+				"should run without constraints",
+				"should run with version in range [2.0.0, ~)",
+				"should run with version in range [2.0.0, 3.0.0)",
+				"should run with version in range [2.0.0, 4.0.0)",
+				"should run with version in range [2.0.0, 5.0.0)",
+				"should inherit container constraint",
+				"should narrow down the constraint",
+				"shouldn't expand the constraint",
+				"should run without constraints by table driven",
+				"should run with version in range [2.0.0, ~) by table driven",
+			}
+			skippedSpecs := []string{
+				"shouldn't run with version in a conflict range",
+				"shouldn't combine with a conflict constraint",
+				"shouldn't run with version in a conflict range by table driven",
+			}
+			Ω(specs).Should(HaveLen(len(passedSpecs) + len(skippedSpecs)))
```

Comment:
i think we need a stronger test here.  You could harvest the leaf node texts and use Gomega’s `ConsistsOf()` to help you match the strings.  Alternatively you could iterate through every spec from the json file and confirm that it’s semver version passes the filter.